sycl/doc/extensions/DiscardQueueEvents/SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS.asciidoc

```diff
@@ -0,0 +1,187 @@
+by using this property the application informs a SYCL implementation that the returned event can be discarded.
+
+In cases where it will be possible, the implementation makes a decision about internal optimization -- avoiding
+the creation of low-level event from the backend. The property affects only kernel cases,
+i.e. avoid creating low-level event for submission of kernels.
+Performance improvement is the main motivation for using this property.
+
+Below is a usage example:
+[source,c++]
+----
+  sycl::property_list props{ext::oneapi::property::queue::discard_events{}
+                            property::queue::in_order{}};
+  sycl::queue Queue( props );
+
+  // some USM preparations ..
+
+  sycl::event e1, e2, e3;
+
+  // returning "invalid" events from each submission function:
+  e1 = Queue.parallel_for(NDRange, [=](nd_item<1> item){ do_smth1(); });
+
+  e2 = Queue.single_task([=](){ do_smth2(); });
+
+  e3 = Queue.submit([&](handler &CGH) { CGH.parallel_for(NDRange, [=](nd_item<1> item){ do_smth3(); }); });
+
+  Queue.wait();
+----
+
+In the example above, the application doesn't use sycl events: `e1`, `e2`, `e3`
+and is waiting for the end of work by `queue::wait()`. The returned events will be
+_invalid_ events. _invalid_ event is a `sycl::event` with a status `invalid` and the event returns from each kernel
+submission when queue is created with the `discard_events` property.
+See the description of behavior for this event below for details.
+
+Here, only those methods for the _invalid_ event are described that have behavior different from the default event behavior:
+[source,c++]
+----
+// must throw an exception with the errc::invalid error code.
+void wait();
+
+// if invalid event is passed into the function, must throw an exception with the errc::invalid error code.
+static void wait(const std::vector<event> &eventList);
+
+// must throw an exception with the errc::invalid error code.
+void wait_and_throw();
+
+// if invalid event is passed into the function, must throw an exception with the errc::invalid error code.
+static void wait_and_throw(const std::vector<event> &eventList);
+
+// must return info::event_command_status::ext_oneapi_unknown
+get_info<info::event::command_execution_status>() const;
+----
+
+The behavior when _invalid_ event is passed into handler API:
+[source,c++]
+----
+// must throw an exception with the errc::invalid error code.
+handler::depends_on(event Event)
+
+// must throw an exception with the errc::invalid error code.
+handler::depends_on(const std::vector<event> &Events)
+----
+
+== DPCPP implementation limitations
+
+The `discard_events` property should be taken as a hint for SYCL implementation.
+In all kernel cases, SYCL implementation returns _invalid_ event, regardless of whether the optimization was done.
+Optimization occurs if the queue has `discard_events` property and all features presented below are not met:

```

Comment:
I think the title of this section could be improved.  The wording "implementation limitations" sounds too negative.  Maybe "Optimization behavior for DPC++"?  We should also make it clear that this is a non-normative part of the spec.  I'd suggest wording like this for the initial paragraph:

> This non-normative section describes the conditions when the DPC++ implementation provides an optimization benefit for the `discard_events` property.
>
> * The queue must be constructed with the `in_order` property.
> * A kernel submitted to the queue must not use the fallback assert feature.
> * A kernel submitted to the queue must not use streams or buffer /  image accessors.  However, local accessors do not inhibit optimization.

Reply:
I like "Optimization behavior for DPC++", applied your suggestion.